include/mamba/configuration.hpp

```diff
@@ -0,0 +1,1117 @@
+
+            return true;
+        }
+    };
+
+    template <>
+    struct convert<mamba::ChannelPriority>
+    {
+        static Node encode(const mamba::ChannelPriority& rhs)
+        {
+            if (rhs == mamba::ChannelPriority::kStrict)
+            {
+                return Node("strict");
+            }
+            else if (rhs == mamba::ChannelPriority::kFlexible)
+            {
+                return Node("flexible");
+            }
+            else if (rhs == mamba::ChannelPriority::kDisabled)
+            {
+                return Node("disabled");
+            }
+            else
+            {
+                return Node();
+            }
+        }
+
+        static bool decode(const Node& node, mamba::ChannelPriority& rhs)
+        {
+            if (!node.IsScalar())
+            {
+                return false;
+            }
+
+            auto str = node.as<std::string>();
+
+            if (str == "strict")
+            {
+                rhs = mamba::ChannelPriority::kStrict;
+            }
+            else if ((str == "flexible") || (str == "true"))
+            {
+                rhs = mamba::ChannelPriority::kFlexible;
+            }
+            else if (str == "disabled")
+            {
+                rhs = mamba::ChannelPriority::kDisabled;
+            }
+            else
+            {
+                return false;
+            }
+
+            return true;
+        }
+    };
+
+    template <>
+    struct convert<fs::path>
+    {
+        static Node encode(const fs::path& rhs)
+        {
+            return Node(rhs.string());
+        }
+
+        static bool decode(const Node& node, fs::path& rhs)
+        {
+            if (!node.IsScalar())
+            {
+                return false;
+            }
+
+            rhs = fs::path(node.as<std::string>());
+            return true;
+        }
+    };
+}
+
+
+namespace mamba
+{
+    class ConfigurableInterface;
+
+    namespace detail
+    {
+        template <class T>
+        struct cli_config
+        {
+            using storage_type = T;
+
+            cli_config(const storage_type& value)
+                : m_value(value){};
+
+            bool defined()
+            {
+                return true;
+            };
+            T value()
+            {
+                return m_value;
+            };
+
+            storage_type m_value;
+        };
+
+
+        template <>
+        struct cli_config<bool>
+        {
+            using storage_type = int;
+
+            cli_config(const int& value)
+                : m_value(value){};
+
+            bool defined()
+            {
+                return m_value != 0;
+            };
+            bool value()
+            {
+                if (m_value == 1)
+                {
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+                throw std::runtime_error("Should not happen!");
```

Comment:
This runtime error can never be hit, so might as well remove it. Or remove the else branch and replace it with return false.

Reply:
nice catch, it was intended to be `else if(m_value == -1)`